src/sentry/static/sentry/app/views/settings/team/teamProjects.jsx

```diff
@@ -0,0 +1,131 @@
+import PanelHeader from '../components/panelHeader';
+
+import {sortProjects} from '../../../utils.jsx';
+import {t} from '../../../locale';
+
+const TeamProjects = createReactClass({
+  displayName: 'TeamProjects',
+  mixins: [
+    ApiMixin,
+    OrganizationState,
+    Reflux.listenTo(ProjectsStore, 'onProjectUpdate'),
+    Reflux.listenTo(TeamStore, 'onTeamUpdate'),
+  ],
+
+  getInitialState() {
+    let team = TeamStore.getBySlug(this.props.params.teamId);
+    return {
+      allProjects: ProjectsStore.getAll(),
+      error: false,
+      projectListLinked: team ? team.projects : [],
+    };
+  },
+
+  componentWillReceiveProps(nextProps) {
+    let params = this.props.params;
+    if (
+      nextProps.params.teamId !== params.teamId ||
+      nextProps.params.orgId !== params.orgId
+    ) {
+      this.setState(this.getInitialState());
+    }
+  },
+
+  onProjectUpdate() {
+    this.setState({
+      allProjects: ProjectsStore.getAll(),
+    });
+  },
+
+  onTeamUpdate() {
+    this.setState({
+      projectListLinked: TeamStore.getBySlug(this.props.params.teamId).projects,
+    });
+  },
+
+  linkProject(project, value) {
+    let {orgId, teamId} = this.props.params;
+    this.api.request(`/projects/${orgId}/${project.slug}/teams/${teamId}/`, {
+      method: value === 'Add' ? 'POST' : 'DELETE',
+      success: () => {
+        let team = TeamStore.getBySlug(this.props.params.teamId);
+        if (value == 'Add') {
+          team.projects = [...team.projects, project];
+        } else {
+          team.projects = team.projects.filter(({id}) => id != project.id);
+        }
+        TeamActions.updateSuccess(0, teamId, team);
+      },
+      error: e => {
+        IndicatorStore.addError("Wasn't able to change project association.");
+      },
+    });
+  },
+
+  projectPanelcontents(projects, direction) {
+    return sortProjects(projects).map((project, i) => (
+      <PanelItem key={i} align="center">
+        <Box w={1 / 2} p={2}>
+          <ProjectListItem project={project} organization={this.context.organization} />
+        </Box>
+        <Box w={1 / 2} p={2} style={{textAlign: 'right'}}>
+          <Button
+            size="small"
+            className="pull-right"
+            onClick={() => {
+              this.linkProject(project, direction);
+            }}
+          >
+            {direction}
+          </Button>
+        </Box>
+      </PanelItem>
+    ));
+  },
+
+  render() {
+    if (this.state.error) return <LoadingError onRetry={this.fetchData} />;
+
+    let {projectListLinked, allProjects} = this.state;
+
+    let linkedProjects = allProjects.filter(p =>
+      projectListLinked.find(l => l.id === p.id)
+    );
+
+    let otherProjects = allProjects.filter(
+      p => !projectListLinked.find(l => l.id === p.id)
+    );
+
+    return (
+      <div>
+        <Panel>
+          <PanelHeader>{t('Associated Projects:')}</PanelHeader>
+          {this.projectPanelcontents(linkedProjects, 'Remove')}
```

Comment:
I think this should be wrapped with a `<PanelBody>`

nit: Let's use `render*` prefix when returning components